frontend/packages/console-shared/src/utils/pod-ring-utils.ts

```diff
@@ -44,6 +50,29 @@ export const podRingLabel = (obj: K8sResourceKind, canScale: boolean): PodRingLa
   };
 };
 
+export const usePodScalingAccessStatus = (
+  obj: K8sResourceKind,
+  resourceKind: K8sKind,
+  pods: ExtPodKind[],
+  enableScaling?: boolean,
+  impersonate?,
+) => {
+  const [editable, setEditable] = React.useState(false);
+  React.useEffect(() => {
+    checkPodEditAccess(obj, resourceKind, impersonate)
+      .then((resp: SelfSubjectAccessReviewKind) =>
+        setEditable(_.get(resp, 'status.allowed', false)),
+      )
+      .catch((error) => {
+        throw error;
```

Comment:
I'm not sure about modal because the user is not doing anything in this case, we are just checking for edit access. There is a possibility that this error will occur on load showing a modal with error on load would seem like a broken experience and on top of that, we check this for each resource.